workloads/network-perf/common.sh

```diff
@@ -43,29 +43,39 @@ export_defaults() {
   export pin=true
   export networkpolicy=${NETWORK_POLICY:=false}
   export multi_az=${MULTI_AZ:=true}
+  export baremetalCheck=$(oc get infrastructure cluster -o json | jq .spec.platformSpec.type)
   zones=($(oc get nodes -l node-role.kubernetes.io/workload!=,node-role.kubernetes.io/worker -o go-template='{{ range .items }}{{ index .metadata.labels "topology.kubernetes.io/zone" }}{{ "\n" }}{{ end }}' | uniq))
 
-  # If multi_az we use one node from the two first AZs
-  if [[ ${multi_az} == "true" ]]; then
-    # Get AZs from worker nodes
-    log "Colocating uperf pods in different AZs"
-    if [[ ${#zones[@]} -gt 1 ]]; then
-      export server=$(oc get node -l node-role.kubernetes.io/worker,node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",topology.kubernetes.io/zone=${zones[0]} -o jsonpath='{range .items[*]}{ .metadata.labels.kubernetes\.io/hostname}{"\n"}{end}' | head -n1)
-      export client=$(oc get node -l node-role.kubernetes.io/worker,node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",topology.kubernetes.io/zone=${zones[1]} -o jsonpath='{range .items[*]}{ .metadata.labels.kubernetes\.io/hostname}{"\n"}{end}' | tail -n1)
-    else
-      log "At least 2 worker nodes placed in different topology zones are required"
-      exit 1
-    fi
-  # If multi_az is disabled we use the two first nodes from the first AZ
+  #If using baremetal we use different query to find worker nodes
+  if [[ "${baremetalCheck}" == '"BareMetal"' ]]; then
+  log "Colocating uperf pods for baremetal"
+  export server=$(oc get nodes --no-headers -l node-role.kubernetes.io/worker | awk 'NR==1{print $1}')
+  export client=$(oc get nodes --no-headers -l node-role.kubernetes.io/worker | awk 'NR==2{print $1}')
```

Comment:
Do you need a check to ensure atleast two nodes are present?

Reply:
I will write new logic to check at least two are present before entering this part of the code

Reply:
I have updated to check for at least two nodes before continuing to set client/server nodes. Application will terminate gracefully with log output that there are not 2 workers nodes available if it does not see at least 2 minimum.